site/src/components/GitAuth/GitAuth.tsx

```diff
@@ -0,0 +1,115 @@
+import Button from "@material-ui/core/Button"
+import FormHelperText from "@material-ui/core/FormHelperText"
+import { makeStyles, Theme } from "@material-ui/core/styles"
+import { SvgIconProps } from "@material-ui/core/SvgIcon"
+import Tooltip from "@material-ui/core/Tooltip"
+import GitHub from "@material-ui/icons/GitHub"
+import * as TypesGen from "api/typesGenerated"
+import { AzureDevOpsIcon } from "components/Icons/AzureDevOpsIcon"
+import { BitbucketIcon } from "components/Icons/BitbucketIcon"
+import { GitlabIcon } from "components/Icons/GitlabIcon"
+import { Typography } from "components/Typography/Typography"
+import { FC } from "react"
+
+export interface GitAuthProps {
+  type: TypesGen.GitProvider
+  authenticated: boolean
+  authenticateURL: string
+  error?: string
+}
+
+export const GitAuth: FC<GitAuthProps> = ({
+  type,
+  authenticated,
+  authenticateURL,
+  error,
+}) => {
+  const styles = useStyles({
+    error: typeof error !== "undefined",
+  })
+
+  let prettyName: string
+  let Icon: (props: SvgIconProps) => JSX.Element
+  switch (type) {
+    case "azure-devops":
+      prettyName = "Azure DevOps"
+      Icon = AzureDevOpsIcon
+      break
+    case "bitbucket":
+      prettyName = "Bitbucket"
+      Icon = BitbucketIcon
+      break
+    case "github":
+      prettyName = "GitHub"
+      Icon = GitHub
+      break
+    case "gitlab":
+      prettyName = "GitLab"
+      Icon = GitlabIcon
+      break
+    default:
+      throw new Error("invalid git provider: " + type)
+  }
+
+  return (
+    <Tooltip
+      title={
+        authenticated ? "You're already authenticated! No action needed." : ``
```

Comment:
Double-checking: do we need a re-authenticate workflow?

Reply:
You can still right click to open in a new tab if need-be, but we verify with the external endpoints that the token is valid, so there _shouldn't be_ a case where the token needs to be revalidated.

Reply:
Perfect, thanks for explaining!